### ISSUE TITLE
RollupJobTaskTests suite timing out

### ISSUE THREAD

**Issue by davidkyle** [elastic/elasticsearch#57411]
**Build scan**:
https://gradle-enterprise.elastic.co/s/lei3dbypn5mjg
https://gradle-enterprise.elastic.co/s/bhdgqmlqfk3ii

**Repro line**:
 ./gradlew ':x-pack:plugin:rollup:test' --tests "org.elasticsearch.xpack.rollup.job.RollupJobTaskTests" \
  -Dtests.seed=B0D0D9DF682DBEA3 \
  -Dtests.security.manager=true \
  -Dtests.locale=en-US \
  -Dtests.timezone=UTC \
  -Druntime.java=8

**Reproduces locally?**:
No

**Applicable branches**:
7.8

**Failure history**:
2 failures in the last 7 days

**Failure excerpt**:
```
com.carrotsearch.randomizedtesting.UncaughtExceptionError: Captured an uncaught exception in thread: Thread[id=33, name=elasticsearch[test][generic][T#2], state=RUNNABLE, group=TGRP-RollupJobTaskTests]
Caused by: java.lang.AssertionError: Should not have timed out
	at __randomizedtesting.SeedInfo.seed([B0D0D9DF682DBEA3]:0)
	at org.junit.Assert.fail(Assert.java:88)
	at org.elasticsearch.xpack.rollup.job.RollupJobTaskTests$25.doExecute(RollupJobTaskTests.java:859)
	at org.elasticsearch.client.support.AbstractClient.execute(AbstractClient.java:399)
	at org.elasticsearch.client.FilterClient.doExecute(FilterClient.java:65)
	at org.elasticsearch.client.ParentTaskAssigningClient.doExecute(ParentTaskAssigningClient.java:63)
	at org.elasticsearch.client.support.AbstractClient.execute(AbstractClient.java:399)
	at org.elasticsearch.xpack.core.ClientHelper.executeAsyncWithOrigin(ClientHelper.java:92)
	at org.elasticsearch.xpack.core.ClientHelper.executeWithHeadersAsync(ClientHelper.java:155)
	at org.elasticsearch.xpack.rollup.job.RollupJobTask$ClientRollupPageManager.doNextSearch(RollupJobTask.java:114)
	at org.elasticsearch.xpack.core.indexing.AsyncTwoPhaseIndexer.triggerNextSearch(AsyncTwoPhaseIndexer.java:581)
	at org.elasticsearch.xpack.core.indexing.AsyncTwoPhaseIndexer.nextSearch(AsyncTwoPhaseIndexer.java:567)
	at org.elasticsearch.xpack.core.indexing.AsyncTwoPhaseIndexer.lambda$maybeTriggerAsyncJob$6(AsyncTwoPhaseIndexer.java:225)
	at org.elasticsearch.action.ActionListener$1.onResponse(ActionListener.java:63)
	at org.elasticsearch.xpack.rollup.job.RollupIndexer.onStart(RollupIndexer.java:121)
	at org.elasticsearch.xpack.core.indexing.AsyncTwoPhaseIndexer.lambda$maybeTriggerAsyncJob$7(AsyncTwoPhaseIndexer.java:222)
	at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:636)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
```

**Comment by iverase** [comment]
Another occurrence:

https://gradle-enterprise.elastic.co/s/uyaop4dwf6svm

**Comment by ywelsch** [comment]
Another occurrence:

https://gradle-enterprise.elastic.co/s/jdfuylgtur5pi

Can someone pick this up?

**Comment by polyfractal** [comment]
I'll take a look.

**Comment by cbuescher** [comment]
Another one today on 7.8: https://gradle-enterprise.elastic.co/s/iacrqhs6a2lwe

**Comment by probakowski** [comment]
And another one https://gradle-enterprise.elastic.co/s/sgymbr4bsmlny

**Comment by polyfractal** [comment]
Seems to only be failing on 7.8.  Digging a bit and I think we just need to backport https://github.com/elastic/elasticsearch/pull/56438 further.  I'll take care of that now

**Comment by polyfractal** [comment]
I backported the fix in 07236558cde40183ca0b3ff6169723002dafd895, going to close this issue... if it comes back it was something else :)
